src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1140,23 +1140,34 @@ LogicalResult ONNXErfOp::inferShapes(
 // PowOp
 //===----------------------------------------------------------------------===//
 
-LogicalResult ONNXPowOp::inferShapes(
-    std::function<void(mlir::Region &)> doShapeInference) {
-  if (!getOperand(0).getType().isa<RankedTensorType>() ||
-      !getOperand(1).getType().isa<RankedTensorType>())
-    return success();
-  RankedTensorType lhsTy = getOperand(0).getType().cast<RankedTensorType>();
-  RankedTensorType rhsTy = getOperand(1).getType().cast<RankedTensorType>();
+static LogicalResult verify(ONNXPowOp op) {
+  ONNXPowOp operandAdaptor = ONNXPowOp(op);
+  
+  RankedTensorType lhsTy = operandAdaptor.X().getType().cast<RankedTensorType>();
+  RankedTensorType rhsTy = operandAdaptor.Y().getType().cast<RankedTensorType>();
```

Comment:
The type for input X and Y could be tensor or memref. The type is tensor when shape inference is called. When you move the check out to verifier, the memref type has to be taken into consideration.
RankedTensorType and MemRef are subclass of ShapedType. You should cast the type to ShapedType and then getElementType() to handle both.